src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1804,6 +1938,38 @@ def nic_attach_choices(self):
         """
         return self.middleware.call_sync('interface.choices', {'exclude': ['epair', 'tap', 'vnet']})
 
+    @accepts()
+    def pptdev_choices(self):
+        """
+        Available choices for PCI passthru device.
+        """
+        if not self.pptdevs:
+            proc = None
+            try:
+                proc = subprocess.Popen(['/usr/sbin/pciconf', '-l'],
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.DEVNULL, text=True)
+                outs, errs = proc.communicate(timeout=5)
+            except (subprocess.TimeoutExpired, OSError, ValueError) as e:
+                if proc is not None:
+                    proc.kill()
+                    outs, errs = proc.communicate()
+                self.logger.warn(f'An error occured when checking the PCI configuration '
```

Comment:
Please see the logging message below. Also while we are here can we please make sure to only use `f-strings` where required ?

Reply:
Not sure I understand your reference to "the logging message below". Do you refer to using "self.middleware.logger.error" or are suggesting some rewording of the message (in addition to changing the . to :)?